### ISSUE TITLE
Sensor process get sleeped

### ISSUE THREAD

**Issue by lalo-uy** [jangevaare/cbpi-OneWireAdvanced#6]
My setup consist of 8 DS18B20 on a bus.
For a second time this week a sensor process keep sleeped while the rest of the app runs ok.
Not the same sensor.
The sensor is readable on the filesystem.
You can see in the log that it stop updating al 15:33
Upon restart goes ok.

2018-07-10 15:31:13,6.0
2018-07-10 15:31:14,6.0
2018-07-10 15:31:22,5.937
2018-07-10 15:32:08,6.0
2018-07-10 15:32:24,5.937
2018-07-10 15:32:34,5.937
2018-07-10 15:33:03,5.937
2018-07-10 15:33:09,6.0
2018-07-10 15:33:21,6.0
2018-07-10 15:33:40,5.937
2018-07-10 15:33:52,6.0
2018-07-11 14:11:04,-0.375
2018-07-11 14:11:12,-0.375
2018-07-11 14:11:20,-0.375
2018-07-11 14:11:28,-0.375
2018-07-11 14:11:36,-0.375
2018-07-11 14:11:44,-0.375
2018-07-11 14:11:52,-0.375
2018-07-11 14:12:00,-0.375
2018-07-11 14:12:08,-0.375

Is there a way to check the sleeped threads ?

![f5](https://user-images.githubusercontent.com/11410289/42590027-1424916c-8519-11e8-8b07-2b447f47409c.png)

**Comment by jangevaare** [comment]
Hello, 
Are you using the latest version? The most recent commit a couple months ago, bdf93b0a7742ac9dc3d366dc32649d801a640c54 did away with initiating a subprocess (within the plugin anyways) for getting a temperature reading. I can't speak to the threads CBPi may or may not implement beyond the scope of this plugin however.

If you want to receive notifications when a temperature reading is filtered, or can not be completed in the update interval (one of which I expect is your issue), make sure the notification duration is long enough for you to catch it. Either way you should see messages in the log if the plugin ran into issues - one of the two below:

https://github.com/jangevaare/cbpi-OneWireAdvanced/blob/master/__init__.py#L185

https://github.com/jangevaare/cbpi-OneWireAdvanced/blob/master/__init__.py#L196

**Comment by lalo-uy** [comment]
Yes , I have the last version.
No msg in the log since I have raised de logging to Errors. Got to many
Info msg from de Ui.
Will change yours to Error level.
Any way that don't cause a while() freeze.
I suspect some exception ocurred after a long time running. Uptime is 8
days.

**Comment by jangevaare** [comment]
The plugin only stores a single "last temperature" between loops, and a "warning count" so there's nothing in it that would change with a long running duration (if you find different in the code, let me know, but it's a pretty simple plugin). I can't speak to the overall reliability of CBPi or your wiring of DS18B20 sensors. In my experience communication with these sensors is very sensitive to electromagnetic interference, and for the high reliability you are looking for it is essential that best practices are used for your control panel and sensors.

Anyways, turn on the logging for this plugin if you're trying to narrow down your issue - I hope that can help. For the number of sensors you have you may also have to increase the update interval as well.

**Comment by lalo-uy** [comment]
Yes, I have raised the interval to 15s . Make no sens on a 60L fermenter
any lower than a minute anyway.
My installation is anything but optimal. The CBPi is next to the Kettles
and the ferneters are 20m away.
To make thing worst one of the pumps in the kettle have a VFD that when
running pops a lot of warnings from your plug in.
This is not critical because it runs a short time and only the brewing day.

Will keep you posted.

**Comment by jangevaare** [comment]
At 15 seconds I wouldn't think you'd have any issues. I thought maybe you were <2 seconds, as some people try to push it as far as they can this way.

I use a VFD with my system as well and it gave me a lot of temperature sensor issues at first. I had to address that with an EMI filter and better shielding. Now everything is good on my end. 

BTW: I actually don't use CBPi anymore - I use a [custom dashboard made with node-red](https://github.com/jangevaare/node-red-brew2) .

**Comment by jangevaare** [comment]
I'm going to close this - feel free to reopen if you have reason to believe it's an issue with this plugin's code. Hope you've managed to sort out!

**Comment by lalo-uy** [comment]
No more problems with 15s refresh.
Any way will add a hold option to be used while rotating logs.

**Comment by lalo-uy** [comment]
I don't know how to reopen this.
Still with problems with this.
Sensor value freezes and no more updates in that sensor log file.
Try to recover with: 
`curl -X POST http://10.9.0.109:5000/api/sensor/1/action/stop

 curl -X POST http://10.9.0.109:5000/api/sensor/1/action/init
`
but no luck.
The only way is opening the sensor in HW profile and click Update
Occurs on random sensors at random times.

**Comment by jangevaare** [comment]
I'm sorry, I'm stumped as to what this could be. I don't see anything within the plugin code that would result in this - pretty basic plugin. If it is software, I would wager on it being an issue with CBPi itself, otherwise I'd guess intermittent hardware issue.

Have you looked to see if there's any log messages? This plugin does send some messages to the app logger if it encounters problems.

Have you tried the standard onewire plugin? Do you have the same issues with it? That would be my next suggestion.

**Comment by lalo-uy** [comment]
Yes, have put standard 1W plugin and have 1k or 2k spikes temps from time
to time.
The ones I want to filter  ;)
Obvious 1 bit error in tx.
The stranger is that the errors are on the nearest probes.

**Comment by lalo-uy** [comment]
Will add lot of debug messages .
If it is a hardware problem should give error but not cancel.
Other option is a background task that check and restart the sensor

**Comment by jangevaare** [comment]
If you are inclined to try this again @lalo-uy , I created a new branch (untested), that puts the temperature reading function in a try block.

https://github.com/jangevaare/cbpi-OneWireAdvanced/tree/trycatch

**Comment by lalo-uy** [comment]
OK, will do it a run and keep you posted
